src/HotChocolate/Core/test/Types.Tests/Types/Relay/NodeTypeTests.cs

```diff
@@ -371,4 +408,59 @@ public class Query9
         public Qux GetBarById(string id)
             => new Qux(id);
     }
+
+    public class Query10
+    {
+        [NodeResolver]
+        public IDocument GetDocumentById(string id)
+            => new DrivingLicense(id);
+
+        public interface IDocument
+        {
+            public string Id { get; }
+        }
+
+        [Node]
+        public record DrivingLicense(string Id) : IDocument;
+    }
+
+    public class Query11
+    {
+        [NodeResolver]
+        public Document GetDocumentById(string id)
+            => new DrivingLicense(id);
+
+        // TODO: Will be nice to have it as interface in schema
```

Comment:
// TODO: Will be nice to have it as interface in schema

Reply:
I think I will remove this TODO and maybe create an issue. The idea is to represent abstract classes as interface on the schema. Currently this can be achieved by creating explicit types (see the test with `Query12`)